### ISSUE TITLE
Path in no_std

### ISSUE THREAD

**Issue by ogoffart** [slint-ui/slint#4178]
Update: the Path is now supported, but not yet with no_std.

-----

The `Path` element is not supported with the software renderer.

We have different choice to implement it:
 1. use some third party crate such as tiny_skia or something to draw the path in an cached image.  The drawback is that this will require `std`, but should be fairly easy to implement.
 2. Make it in a way that could work with `no_std` and with minimal allocation on MCUs

**Comment by ogoffart** [comment]
https://github.com/slint-ui/slint/pull/9894 Implemented the feature, but did not yet make a public feature flag that can be enabled

**Comment by ogoffart** [comment]
This is now enabled by default with `std`.
Enabling it with `no_std` could work, but `Path` is disabled for no_std in i_slint_core.

**Comment by yanshay** [comment]
> This is now enabled by default with `std`. Enabling it with `no_std` could work, but `Path` is disabled for no_std in i_slint_core.

So it could work but not enabled? Any way to enable it?
Is `std` required or just `alloc` ?

**Comment by ogoffart** [comment]
`std` is currently required because the internal `i_slint_core::items::Path` is gated by it.

It should be possible to change that and make it an optional feature instead (because it also would bring the lyon dependency.)

That shouldn't be a big change.

**Comment by yanshay** [comment]
since you marked it as `good first issue` I tried to get it working with no_std:
- Added feature flag `no_std_path`. 
- configured them to enable places in the code that were gated only behind std
- Added the required lyon  dependencies and auto_enums.
- Configured those without default_features
- Enabled default features when std so will work as usual in that case

But still got errors in  arrayvec and num_traits. 

Seems they get the default features from lyon_extra -> lyon_path. lyon_extra doesn't have a no_std option so I'm not sure this is doable. 
This PR is to solve that: https://github.com/nical/lyon/pull/941

The errors with num_traits were less clear, so not sure about them but it seems like also coming from lyon_extra.

**Comment by ogoffart** [comment]
Right. Maybe the feature to parse commands string at runtime (dynamic string) could be still gated for `std`. And then the lyon parser shouldn't be needed.

**Comment by ogoffart** [comment]
Done

**Comment by yanshay** [comment]
Got a notification on this.
I just saw lyon_extra PR to support no_std was merged there. So maybe the dynamic commands can be supported now as well.

**Comment by Dominaezzz** [comment]
Has anyone successfully used this on no_std yet?

**Comment by sindrehan** [comment]
Yes, it's been running flawlessly on my RP2350 project.
